Add Label Studio Documentation (#746)
huggingface/hub-docs: docs/hub/spaces-sdks-docker-label-studio.md

# 🟧 Label Studio on Spaces

[Label Studio](https://labelstud.io) is an [open-source data labeling
platform](https://github.com/heartexlabs/label-studio) for labeling,
annotating, and exploring many different data types. Additionally, Label Studio
includes a powerful [machine learning
interface](https://labelstud.io/guide/ml.html) that can be used for new model
training, active learning, supervised learning, and many other training
techniques.

> In this guide, you will learn how to deploy Label Studio to user for data

Reviewer: ```suggestion
This guide will teach you how to deploy Label Studio for data
```

labeling and annotation within the Hugging Face Hub. You can use the default
configuration of Label Studio as a self-contained application hosted completely
on the Hub using Docker for demonstration and evaluation purposes, or you can

> attach your own database and cloud storage to host a fully-featured an

Reviewer: ```suggestion
attach your own database and cloud storage to host a fully-featured
```

production-ready application hosted on Spaces.

## ⚡️ Deploy Label Studio on Spaces

You can deploy Label Studio on Spaces with just a few clicks:

<a  href="https://huggingface.co/new-space?template=LabelStudio/LabelStudio">
  <img src="https://huggingface.co/datasets/huggingface/badges/raw/main/deploy-to-spaces-lg.svg" />
</a>

Spaces requires you to define:

* An **Owner**: either your personal account or an organization you're a
  part of. 

* A **Space name**: the name of the space within the account
  you're creating the space.

* The **Visibility**: _private_ if you want the

>   space to be visible only to you or you organization, or _public_ if you want

Reviewer: ```suggestion
  Space to be visible only to you or your organization, or _public_ if you want
```

  it to be visible to other users or applications using the Label Studio API
  (suggested).

## 🚀 Using the Default Configuration

> By default, Label Studio is installed to spaces with a configuration that uses

Reviewer: ```suggestion
By default, Label Studio is installed in Spaces with a configuration that uses
```

local storage for the application database to store configuration, account

> credentials, project information. Labeling tasks and data items are also held

Reviewer: ```suggestion
credentials, and project information. Labeling tasks and data items are also held
```

in local storage. 

<Tip warning={true}>
Storage in Hugging Face Spaces is ephemeral, and data you store in the default

> configuration can be lost in a reboot or reset of the space. Because of this,

Reviewer: ```suggestion
configuration can be lost in a reboot or reset of the Space. Because of this,
```

we strongly encourage you to use the default configuration only for testing and
demonstration purposes.
</Tip>

After launching Label Studio, you will be presented with the standard login

> screen. You can get started by creating a new account using your email address,

Reviewer: ```suggestion
screen. You can start by creating a new account using your email address and
```

> then logging in with your new credentials. Periodically after you log in, Label

Reviewer: ```suggestion
logging in with your new credentials. Periodically after logging in, Label
```

> Studio will give you warning that the storage is ephemeral and data could be

Reviewer: ```suggestion
Studio will warn you that the storage is ephemeral and data could be
```

> lost if your space is restarted. You will also be preseted with a prompt from

Reviewer: ```suggestion
lost if your Space is restarted. You will also be preset with a prompt from
```

Heidi, the helpful Label Studio mascot, to create a new project to start
labeling your data. To get started, check out the Label Studio ["Zero to One"
tutorial](https://labelstud.io/blog/introduction-to-label-studio-in-hugging-face-spaces/)

> with a guide on how to build a annotation interface for sentiment analysis. 

Reviewer: ```suggestion
with a guide on how to build an annotation interface for sentiment analysis. 
```

## 🛠️ Configuring a Production-Ready Instance of Label Studio

> To make your space production-ready, you will need to make three configuration

Reviewer: ```suggestion
To make your Space production-ready, you will need to make three configuration
```

changes:

* Disable the unrestricted creation of new accounts.

* Enable persistence by attaching an external database.

* Attach cloud storage for labeling tasks.

### Disable Unrestricted Creation of New Accounts

The default configuration on Label Studio allows for the unrestricted creation

> of new accounts for anyone who has the URL for your applicaiton. You can

Reviewer: ```suggestion
of new accounts for anyone who has the URL for your application. You can
```

[restrict signups](https://labelstud.io/guide/signup.html#Restrict-signup-for-local-deployments)

> by adding the following configuration secrets to your space **Settings**.

Reviewer: ```suggestion
by adding the following configuration secrets to your Space **Settings**.
```

* `LABEL_STUDIO_DISABLE_SIGNUP_WITHOUT_LINK`: Setting this value to `true` will
  disable unrestricted account creation. 

* `LABEL_STUDIO_USERNAME`: This is the username of the account that you will

>   use as the first user in your Label Studio space. It should be a valid email

Reviewer: ```suggestion
  use as the first user in your Label Studio Space. It should be a valid email
```

  address.

* `LABEL_STUDIO_PASSWORD`: This is the password that will be associated with
  the first user account.

> Restart the space to apply these setting. The ability to create new accounts

Reviewer: ```suggestion
Restart the Space to apply these settings. The ability to create new accounts
```

> Restart the space to apply these setting. The ability to create new accounts

Reviewer: Note that the Space is usually restarted after new secrets iirc

from the login screen will be disabled. To create new accounts, you will need
to invite new users in the `Organization` settings in the Label Studio
application.

### Enable Configuration Persistence

> By default this space stores all project configuration and data annotations in

Reviewer: ```suggestion
By default, this Space stores all project configuration and data annotations in
```
